### PR TITLE
7221: Added OrderingParty Section completion integration tests

### PR DESCRIPTION
**Pull request by SimonLLawrence**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 1]
```diff
@@ -4,13 +4,18 @@
...
+        | Fred      | Robinson | Fred.robinson@email.com | 12312543212     |
     Given Orders exist
-        | OrderId    | Description   | OrderStatusId | Created    | LastUpdated | LastUpdatedBy                        | OrganisationId                       |
-        | C000014-01 | A Description | 1             | 05/05/2020 | 09/05/2020  | a11a46f9-ce6f-448a-95c2-fde6e61c804a | 4af62b99-638c-4247-875e-965239cd0c48 |
+        | OrderId    | Description   | OrderStatusId | Created    | OrganisationContactEmail | LastUpdated | LastUpdatedBy                        | OrganisationId                       |
+        | C000014-01 | A Description | 1             | 05/05/2020 |                          | 09/05/2020  | 7b195137-6a59-4854-b118-62b39a3101e | 4af62b99-638c-4247-875e-965239cd0c48 |
+        | C000014-02 | A Description | 1             | 05/05/2020 | Fred.robinson@email.com  | 09/05/2020  | 7b195137-6a59-4854-b118-62b39a3101e | 4af62b99-638c-4247-875e-965239cd0c48 |
     And the user is logged in with the Buyer role for organisation 4af62b99-638c-4247-875e-965239cd0c48
 
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:16]: Extra space not needed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 2]
```diff
@@ -4,13 +4,18 @@
...
 @5321
-Scenario: 1. Displaying the order summary, where the order sections are complete
+Scenario: 1. Displaying the order summary, where the order descript section is complete
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:18]: Change this name

Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:18]: Might be worth looking at specflows example, to make the tests more streamlined, an example of this is in the validation specflow tests

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 3]
```diff
@@ -28,31 +33,71 @@ Scenario: 1. Displaying the order summary, where the order sections are complete
...
+        | commencement-date   | incomplete |
+        | associated-services | incomplete |
+        | service-recipients  | incomplete |
+        | catalogue-solutions | incomplete |
+        | additional-services | incomplete |
+        | funding-source      | incomplete |
+    And the order with orderId C000014-02 has a primary contact
+
+@7221
+Scenario: 3. Displaying the order summary, where the ordering party section is incomplete
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:57]: Is this just the same test as above? No1?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 4]
```diff
@@ -28,31 +33,71 @@ Scenario: 1. Displaying the order summary, where the order sections are complete
...
+        | associated-services | incomplete |
+        | service-recipients  | incomplete |
+        | catalogue-solutions | incomplete |
+        | additional-services | incomplete |
+        | funding-source      | incomplete |
+    And the order with orderId C000014-01 does not have a primary contact
+
 @5321
-Scenario: 2. If the order ID does not exist, return not found
+Scenario: 3. If the order ID does not exist, return not found
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:77]: Can you sort out the numbering to these files

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 5]
```diff
@@ -28,31 +33,71 @@ Scenario: 1. Displaying the order summary, where the order sections are complete
...
+        | supplier            | incomplete |
+        | commencement-date   | incomplete |
+        | associated-services | incomplete |
+        | service-recipients  | incomplete |
+        | catalogue-solutions | incomplete |
+        | additional-services | incomplete |
+        | funding-source      | incomplete |
+    And the order with orderId C000014-02 has a primary contact
+
+@7221
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:56]: Dont do as per the task, its the number of what story you are working on

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature (hunk 3 above)
Comment (SimonLLawrence) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:57]: Explicitly called  out the negative case and proved the order was in a state that would not set the section as complete.  But i can delete it

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature (hunk 2 above)
Comment (SimonLLawrence) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature]: changed the tests to remove some of the  checks that were not relevant

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs [hunk 6]
```diff
@@ -62,7 +62,20 @@ public async Task ThenTheOrderSummarySectionsHaveTheFollowingValues(Table table)
 
             actual.Sections.Should().BeEquivalentTo(expected);
         }
-        
+
+        [Then(@"the order Summary Section (.*) has status (.*)")]
+        public async Task ThenTheOrderSummarySectionsHaveTheFollowingValues(string sectionId,string status)
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs:67]: Can you add a space inbetween the comma and string status please

**Pull request merged**